packages/ui-date-time-input/src/DateTimeInput/index.tsx

```diff
@@ -134,6 +135,35 @@ class DateTimeInput extends Component<DateTimeInputProps, DateTimeInputState> {
             .month(this.state.iso.month())
             .year(this.state.iso.year())
         }
+        if (
+          !this.props.defaultValue &&
+          this.props.initialTimeForNewDate &&
+          !this.state?.timeSelectValue
+        ) {
+          const hour = Number(this.props.initialTimeForNewDate.slice(0, 2))
```

Comment:
I think you should validate this with Moment and parse it after so you can extract the hour and minutes safely and reliably

Reply:
This is a simple parse of two numbers here, I added plenty of validation here and in the props check. I consider this enough, thanks for your advice though.

We would like to change Moment for something more future proof as soon as possible.